src/renderers/shaders/ShaderChunk/envmap_vertex.glsl.js

```diff
@@ -19,8 +19,6 @@ export default /* glsl */`
 
 		}
 
-		vec3 worldNormal = inverseTransformDirection( transformedNormal, viewMatrix );
```

Comment:
>The vertex shader also now also computes the world normal only once and reuses it for the envmap and shadow maps.

For pretty-much the same reason as in #18891, I would be inclined to avoid sharing local variables between shader chunks.

This particular code block only applies to PHONG, I think.

Please reconsider creating `shadowWorldNormal` separately.... Is there a true downside to doing that?

Reply:
Ah right, I thought that this code would be a part of more shaders. I agree that computing shadowWorldNormal separately makes more sense.